pkgs/development/libraries/getdns/default.nix

```diff
@@ -1,37 +1,83 @@
-      specification was originally edited by Paul Hoffman. It is intended to make all
-      types of DNS information easily available to application developers and non-DNS
-      experts. DNSSEC offers a unique global infrastructure for establishing and
-      enhancing cryptographic trust relations. With the development of this API the
-      developers intend to offer application developers a modern and flexible
-      interface that enables end-to-end trust in the DNS architecture, and which will
-      inspire application developers to implement innovative security solutions in
-      their applications.
-    '';
-    homepage = "https://getdnsapi.net";
+{ lib, stdenv, fetchurl, cmake, darwin, doxygen, libidn2, libyaml, openssl
+, systemd, unbound, yq }:
+let
+  metaCommon = with lib; {
     maintainers = with maintainers; [ leenaars ehmry ];
     license = licenses.bsd3;
     platforms = platforms.all;
   };
+in rec {
+
+  getdns = stdenv.mkDerivation rec {
+    pname = "getdns";
+    version = "1.7.0";
+
+    src = fetchurl {
+      url = "https://getdnsapi.net/releases/${pname}-${
+          with builtins;
+          concatStringsSep "-" (splitVersion version)
+        }/${pname}-${version}.tar.gz";
+      sha256 = "sha256-6ocTzl4HesdrFBjOtq/SXm1OOelgD29egdOjoTpg9lI=";
+    };
+
+    nativeBuildInputs = [ cmake ];
+
+    buildInputs = [ doxygen libidn2 openssl unbound ];
+
+    meta = with lib;
+      metaCommon // {
+        description = "A modern asynchronous DNS API";
+        longDescription = ''
+          getdns is an implementation of a modern asynchronous DNS API; the
+          specification was originally edited by Paul Hoffman. It is intended to make all
+          types of DNS information easily available to application developers and non-DNS
+          experts. DNSSEC offers a unique global infrastructure for establishing and
+          enhancing cryptographic trust relations. With the development of this API the
+          developers intend to offer application developers a modern and flexible
+          interface that enables end-to-end trust in the DNS architecture, and which will
+          inspire application developers to implement innovative security solutions in
+          their applications.
+        '';
+        homepage = "https://getdnsapi.net";
+      };
+  };
+
+  stubby = stdenv.mkDerivation rec {
+    pname = "stubby";
+    version = "0.4.0";
+    outputs = [ "out" "stubbyExampleJson" ];
+
+    inherit (getdns) src;
+    sourceRoot = "${getdns.name}/stubby";
+
+    nativeBuildInputs = [ cmake libyaml yq ];
+
+    buildInputs = [ doxygen getdns openssl systemd ]
+      ++ lib.optionals stdenv.isDarwin [ darwin.Security ];
+
+    postInstall = ''
+      yq \
+        < $NIX_BUILD_TOP/$sourceRoot/stubby.yml.example \
+        > $stubbyExampleJson
+    '';
+
+    passthru.settingsExample = with builtins;
+      fromJSON (readFile stubby.stubbyExampleJson);
```

Comment:
this is IFD unless I am mistaken. Maybe it would be better to include the yaml file in nixpkgs and check during the build that the upstream version and the nixpkgs version of the file are still in sync.

Reply:
Yes, this is IFD, but it is not evaluated anywhere, only mentioned in the documentation of the NixOS module as literal code.

https://github.com/NixOS/nixpkgs/blob/f9fcf2a2db1d1d14acf8bc9991735580be662f28/nixos/modules/services/networking/stubby.nix#L17

Reply:
ok makes sense.